executor/show.go

```diff
@@ -691,33 +691,38 @@ func (e *ShowExec) fetchShowCreateTable() error {
 	partitionInfo := tb.Meta().Partition
 	if partitionInfo != nil {
 		// this if statement takes care of range columns case
-		if partitionInfo.Columns != nil && partitionInfo.Type == model.PartitionTypeRange {
-			buf.WriteString(fmt.Sprintf("\nPARTITION BY RANGE COLUMNS("))
-			for i, col := range partitionInfo.Columns {
-				buf.WriteString(col.L)
-				if i < len(partitionInfo.Columns)-1 {
-					buf.WriteString(",")
-				}
-			}
-			buf.WriteString(") (\n")
+		if partitionInfo.Type == model.PartitionTypeHash {
```

Comment:
can we use a function to wrap this logic, make code look clear?

Reply:
@winkyao done.